okhttp/src/commonJvmAndroid/kotlin/okhttp3/Response.kt

```diff
@@ -192,6 +192,26 @@ class Response internal constructor(
   @Throws(IOException::class)
   fun trailers(): Headers = trailersSource.get()
 
+  /**
+   * Returns the trailers after the HTTP response, if they are available to read immediately. Unlike
+   * [trailers], this doesn't block if the trailers are not immediately available, and instead
+   * returns null.
+   *
+   * This will typically return null until [ResponseBody.source] has buffered the last byte of the
+   * response body. Call `body.source().request(1024 * 1024)` to block until either that's done, or
+   * 1 MiB of response data is loaded into memory. (You could use any size here, though large values
+   * risk exhausting memory.)
+   *
+   * This returns an empty value if the trailers are available, but have no data.
+   *
+   * It is not safe to call this concurrently with code that is processing the response body.
```

Comment:
Does this include reading the body in chunks? or is it fine to interleave?

I realise this isn't really useful as trailers terminate the stream. But maybe interesting if you think the body has been received, just not consumed yet.